parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -171,6 +171,8 @@ private final STATE error() throws IOException {
 
   private STATE state = STATE.NOT_STARTED;
 
+  private final CodecFactory codecFactory;
```

Comment:
It seems that this is not the right place to put this as this class does not use the codecFactory.
it just creates it and makes it available. I'd rather have private members not accessible to the outside to respect encapsulation and layering of the code.